quinn-proto/src/connection.rs

```diff
@@ -541,7 +541,9 @@ impl Connection {
 
     fn set_key_discard_timer(&mut self, now: Instant) {
         let time = if self.spaces[SpaceId::Handshake as usize].crypto.is_some() {
-            now + self.pto() * 3
+            // Severe packet loss during this period can cause the connection to fail, so allocate
```

Comment:
Maybe add something to the comment about the potential downside of what we're trading off here?

Reply:
Elaborated to the best of my understanding; might be worth double-checking on slack or against TLS references, though.

Reply:
The TLS draft [is very relaxed](https://quicwg.org/base-drafts/draft-ietf-quic-tls.html#discarding-unused-keys) about requirements here; handshake key discard does not appear to actually be mandated at all, though I understand a future draft may be more specific about reasonable behavior here.